### PR TITLE
Tezos hooks with TypeScript

### PR DESCRIPTION
**Pull request by claudebarde**

### CODE REVIEW

File: use-balance.ts [hunk 1]
```diff
@@ -1,30 +1,36 @@
 import { useState, useEffect } from "react";
 import { useTezosContext } from "./TezosContext";
 
-export function useBalanceState(address = "", contractOperationsCount = 0) {
+export function useBalance(address: string = "") {
```
Comment (chiptus) [inline use-balance.ts:4]: shouldn't we have return types for the hooks?

**Review by chiptus — CHANGES_REQUESTED** [review]

File: use-contract.ts [hunk 2]
```diff
@@ -53,6 +63,7 @@ export function useContract(contractAddress: string): ContractHook {
   }
 
   async function loadStorage(contract) {
+    console.log(contract);
```
Comment (chiptus) [inline use-contract.ts:66]: remove console.log

File: use-contract.ts [hunk 3]
```diff
@@ -10,9 +10,19 @@ export function useContract(contractAddress: string): ContractHook {
...
   const [loading, setLoading] = useState(false);
+  const [refreshStorageInterval, setRefreshStorageInterval] = useState(
+    undefined
+  );
 
   useEffect(() => {
     loadStorage(contract);
+    clearInterval(refreshStorageInterval);
+    // sets interval
+    const interval = setInterval(async () => loadStorage(contract), 6000);
```
Comment (chiptus) [inline use-contract.ts:21]: isn't 6 seconds too low? I believe a minute is enough, as the blockchain updates every one minute, no?

File: use-balance.ts [hunk 4]
```diff
@@ -1,9 +1,11 @@
 import { useState, useEffect } from "react";
 import { useTezosContext } from "./TezosContext";
+import { BalanceHook } from "./types";
+import BigNumber from "bignumber.js";
 
-export function useBalance(address: string = "") {
+export function useBalance(address: string = ""): BalanceHook {
   const { tezos } = useTezosContext();
-  const [balance, setBalance] = useState();
+  const [balance, setBalance]: [number, (p: number) => void] = useState();
```
Comment (chiptus) [inline use-balance.ts:8]: useState<number> doesn't work? like in use-contract?

**Review by chiptus — CHANGES_REQUESTED** [review]

File: use-contract.ts (hunk 2 above)
Comment (claudebarde) [reply use-contract.ts]: Fixed

File: use-balance.ts (hunk 4 above)
Comment (claudebarde) [reply use-balance.ts:8]: This only enforces to pass a number for the balance in `setBalance` (and not a string for example). Not absolutely necessary though, just to make TS throw an error if anything else but a number is passed. Now that I think about it, the user doesn't have access to `setBalance` so we can relax it :)

**Review by chiptus — APPROVED** [review]

**Pull request merged**